File: sorting/selection_sort/selection_sort.js
function selection_sort(elements) {
    var length = elements.length;

    for (let i = 0; i < length - 1; i++) {
        var min_element_index = i;

        for (let j = i + 1; j < length; j++) {
            if (elements[j] < elements[min_element_index]) {
                min_element_index = j;
            }
        }

        [elements[min_element_index], elements[i]] = [elements[i], elements[min_element_index]]
    }
    return elements;
}

const input = [3, 7, 2, 9, 0, 6]

console.log(`selection_sort: ${selection_sort(input.slice())}`);

File: sorting/selection_sort/selection_sort.dart
List<int> selectionSort(List<int> elements) {
  int length = elements.length;
  late int temp;
  for (var i = 0; i < length - 1; i++) {
    int minElementIndex = i;
    for (var j = i; j < length; j++) {
      if (elements[j] < elements[minElementIndex]) {
        minElementIndex = j;
      }
    }

    temp = elements[i];
    elements[i] = elements[minElementIndex];
    elements[minElementIndex] = temp;
  }
  return elements;
}

void main() {
  List<int> input = [3, 19, 8, 2, 5, 1,0];
  print(input);
  print("selectionSort ${selectionSort(List<int>.from(input))}");
}
